core/src/main/scala/cromwell/util/GracefulShutdownHelper.scala

```diff
@@ -0,0 +1,34 @@
+package cromwell.util
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Terminated}
+import akka.pattern.GracefulStopSupport
+import cats.data.NonEmptyList
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+
+object GracefulShutdownHelper {
+  case object ShutdownCommand
+}
+
+trait GracefulShutdownHelper extends GracefulStopSupport { this: Actor with ActorLogging =>
+  private [util] var shuttingDown: Boolean = false
```

Comment:
I'm assuming this is for testing but I really don't like the idea of having mutable state in actors exposed at all. Any way around this?